src/components/Table/EditColumnsButton.tsx

```diff
@@ -0,0 +1,91 @@
+interface EditColumnsButtonProps<R extends Kinded, S> extends Pick<OverlayTriggerProps, "trigger" | "placement" | "disabled" | "tooltip"> {
+  columns: GridColumn<R, S>[];
+  setColumns: Dispatch<SetStateAction<Columns<R, S>>>;
+  title?: string;
+  buttonText?: string;
+  // for storybook purposes
+  defaultOpen?: boolean;
+}
+
+export function EditColumnsButton<R extends Kinded, S = {}>(props: EditColumnsButtonProps<R, S>) {
+  const { defaultOpen, disabled, columns, setColumns, trigger, title, buttonText} = props;
+  const state = useMenuTriggerState({ isOpen: defaultOpen });
+  const buttonRef = useRef<HTMLButtonElement>(null);
+  const { menuTriggerProps } = useMenuTrigger({ isDisabled: !!disabled }, state, buttonRef);
+  const tid = useTestIds(
+    props,
+    isTextButton(trigger) ? trigger.label : isIconButton(trigger) ? trigger.icon : trigger.name,
+  );
+
+  function clearSelections(){
+    columns.map((column) => column.visible = false);
+    setColumns((prevState) => ({ ...prevState, visibleColumns: columns.filter((column) => ((column.canHide && column.visible) || !column.canHide)) }));
+  }
+
+  return (
+    <OverlayTrigger {...props} menuTriggerProps={menuTriggerProps} state={state} buttonRef={buttonRef} {...tid}>
+      <div
+        css={{
+          ...Css.df.fdc.bgWhite.py3.px2.maxw(px(380)).$,
+          "&:hover, &:focus": Css.bshHover.$,
+        }}
+      >
+        <div css={Css.gray500.sm.ttu.$}>{title || "Select columns to show"}</div>
+        <div css={Css.df.add({ flexWrap: "wrap" }).gap2.py2.$}>
+          {columns.map((column, i) => {
+            return (
+              column.canHide && 
+              <div css={Css.add({ flex: "40%"}).ttc.$} key={i}>
+                <Checkbox
+                  label={column.name || ""}
+                  selected={!!column.visible}
+                  onChange={(value) => {
+                    column.visible = value;
+                    setColumns((prevState) => ({ ...prevState, visibleColumns: columns.filter((column) => ((column.canHide && column.visible) || !column.canHide)) }));
+                  }}
+                />
+              </div>
+            )
+          })}
+        </div>
+        <div css={Css.smMd.$}>
+          <Button
+            size="sm"
+            variant={"text"}
+            label={buttonText || "Clear selections"}
+            onClick={clearSelections}
+          />
+        </div>
+      </div>
+    </OverlayTrigger>
+  );
+}
+
+export interface Columns<R extends Kinded, S = {}> {
+  allColumns: GridColumn<R, S>[],
+  visibleColumns: GridColumn<R, S>[]
+}
+
+export function useColumns<R extends Kinded, S = {}>(tableColumns: GridColumn<R, S>[]):
+[Columns<R, S>, Dispatch<SetStateAction<Columns<R, S>>>] {
+
+  const initColumns: Columns<R, S> = useMemo(() => {
+    return {
+      allColumns: tableColumns,
+      visibleColumns: tableColumns.filter((column) => ((column.canHide && column.visible) || !column.canHide))
+    }
+  },[tableColumns]);
```

Comment:
**nit:** I think there are some formatting issues in this. Can you try running `yarn format`?